galsim_extra/image_lattice.py

```diff
@@ -0,0 +1,224 @@
+            xsize, ysize
+        """
+        logger.debug('image %d: Building Lattice: image, obj = %d,%d',image_num,image_num,obj_num)
+
+        extra_ignore = [ 'image_pos' ] # We create this below, so on subequent passes, we ignore it.
+        req = { 'sep' : float , 'xsize' : int , 'ysize' : int }
+        opt = { 'rotation' : float , "border_ratio" : float }
+        params = GetAllParams(config, base, req=req, opt=opt, ignore=ignore+extra_ignore)[0]
+
+        # TODO: is this the preferred way of propagating these parameters from the config?
+        self.sep = params["sep"]
+        self.rotation = params.get("rotation", 0)
+        self.border_ratio = params.get("border_ratio", 1)
+
+        size = params.get('size',0)
+        full_xsize = params.get('xsize',size)
+        full_ysize = params.get('ysize',size)
+
+        if (full_xsize <= 0) or (full_ysize <= 0):
+            raise GalSimConfigError(
+                "Both image.xsize and image.ysize need to be defined and > 0.")
+
+        # If image_force_xsize and image_force_ysize were set in config, make sure it matches.
+        if ( ('image_force_xsize' in base and full_xsize != base['image_force_xsize']) or
+             ('image_force_ysize' in base and full_ysize != base['image_force_ysize']) ):
+            raise GalSimConfigError(
+                "Unable to reconcile required image xsize and ysize with provided "
+                "xsize=%d, ysize=%d, "%(full_xsize,full_ysize))
+
+        return full_xsize, full_ysize
+
+
+    def buildImage(self, config, base, image_num, obj_num, logger):
+        """
+        Build an Image consisting of a tiled array of postage stamps.
+
+        Parameters:
+            config:     The configuration dict for the image field.
+            base:       The base configuration dict.
+            image_num:  The current image number.
+            obj_num:    The first object number in the image.
+            logger:     If given, a logger object to log progress.
+
+        Returns:
+            the final image and the current noise variance in the image as a tuple
+        """
+        full_xsize = base['image_xsize']
+        full_ysize = base['image_ysize']
+        wcs = base['wcs']
+
+        dtype = _ParseDType(config, base)
+        full_image = Image(full_xsize, full_ysize, dtype=dtype)
+        full_image.setOrigin(base['image_origin'])
+        full_image.wcs = wcs
+        full_image.setZero()
+        base['current_image'] = full_image
+
+        # Make a list of ix,iy values according to the specified order:
+        rng = np.random.default_rng()  # TODO: how to best handle RNGs with galsim?
+        # TODO: pull lattice vectors from config
+        v1 = np.asarray([1, 0], dtype=float)
+        v2 = np.asarray([np.cos(np.radians(120)), np.sin(np.radians(120))], dtype=float)
+        x_lattice, y_lattice = build_lattice(full_xsize-1, full_ysize-1, self.sep, base["pixel_scale"], v1, v2, self.rotation, self.border_ratio)
```

Comment:
Why the -1's here?

It's a bit confusing to pass `full_xsize-1` as a parameter that will be received as `full_xsize` in the function.  If there is a good reason to subtract 1, then I'd suggest passing these here as just `full_xsize, full_ysize`, and doing the -1 in the function, with a comment explaining why that's a good idea.

Reply:
thanks for catching this and the suggestion -- I do think this is a bug unintentionally carried over from an earlier draft of the code.

Reply:
I've verified that we want to pass `full_xsize` and `full_ysize` as is to `build_lattice` in this implementation